src/amo/components/Collection/index.js

```diff
@@ -207,12 +200,11 @@ export const mapStateToProps = (state: { collections: CollectionsState }) => {
 
 export default compose(
   translate(),
-  withErrorHandler({
-    // This allows to sync the client and the server error handler ids, thus
-    // allowing the client side to be aware of errors thrown on the server.
-    // See: https://github.com/mozilla/addons-frontend/issues/3313
-    id: 'Collection-001',
+  withPageErrorHandler({
     name: 'Collection',
+    extractId: (ownProps: Props) => {
+      return `${ownProps.params.user}/${ownProps.params.slug}`;
```

Comment:
Wow, good catch!

Reply:
> Part of my worry with this approach is that it will be hard to construct the right ID. But maybe we should still try it and see what happens?

I tend to agree, but you also reply to yourself correctly. All in all, the whole error handling feature saves us a lot of time and hides complexity. If the trade-off is to only be sure we have correct IDs, then I think it is a fair price we can afford.